tests/unit/service/test_resourcedriver.py

```diff
@@ -39,19 +40,21 @@ def test_execute(self, logging_context):
         mock_service = MagicMock()
         mock_service.execute_lifecycle.return_value = LifecycleExecuteResponse('123')
         controller = ResourceDriverApiService(service=mock_service)
-        response, code = controller.execute_lifecycle(**{
-            'body': {
-                'lifecycleName': 'Start',
-                'systemProperties': self.__props_with_types({'resourceId': '1', 'b': 1}),
-                'resourceProperties': self.__props_with_types({'a': '2', 'b': 2}),
-                'requestProperties': self.__props_with_types({'reqA': '3', 'reqB': True}),
-                'associatedTopology': [{'id': 'abc', 'name': 'Test', 'type': 'Testing'}],
-                'driverFiles': b'123',
-                'deploymentLocation': {'name': 'test'},
-                'version': '1.0.0'
-            }
-        })
-        mock_service.execute_lifecycle.assert_called_once_with('Start', b'123', {'resourceId': { 'type': 'string', 'value': '1'}, 'b': { 'type': 'integer', 'value': 1} }, {'a': { 'type': 'string', 'value': '2'}, 'b': { 'type': 'integer', 'value': 2}}, {'reqA': {'type': 'string', 'value': '3'}, 'reqB': {'type': 'boolean', 'value': True}}, [{'id': 'abc', 'name': 'Test', 'type': 'Testing'}], {'name': 'test'})
+        app = Flask(__name__)
+        with app.test_request_context():
```

Comment:
Why do we need to introduce a Flask app here now?

Reply:
@dvaccarosenna  We need this as we have added code to fetch the tenant id from request headers inside execute_lifecycle() method. 
`connexion.request.headers['TenantId']`

So if we don't use it then we are getting the below error while executing the test cases which calls execute_lifecycle() method.

```
RuntimeError: Working outside of request context.

This typically means that you attempted to use functionality that needed
an active HTTP request. Consult the documentation on testing for
information about how to avoid this problem.
```

Upon further checking the Flask and Connection Framework related documents I have put the test code inside the Flask context as it was recommended in their documentation for unit tests when we are dealing with any request 

```
app = Flask(__name__)
with app.test_request_context():
  ...existing test code...
```

Reply:
Ok, interesting the `logging_context.set_from_headers` didn't trigger this error as it uses that too. I wonder if we can just use `kwargs.get('request').headers` instead of `connexion.request.headers`?

Reply:
Connexion docs seem to suggest that might not be valid. If not, then we'll leave it as is

Reply:
@dvaccarosenna We are not getting that error in `logging_context.set_from_headers` because we are mocking that in unit test cases. So the actual method `set_from_headers` is not getting executed because of mocking. The test methods are having mocking annotations on top of it. `@patch('ignition.service.resourcedriver.logging_context')` .

Yes, as per the Connexion docs - `"Currently, header parameters are not passed to the handler functions as parameters. But they can be accessed through the underlying connexion.request.headers object which aliases the flask.request.headers object."` So that's why `kwargs.get('request').headers` will not work.

I have even verified it by putting a breakpoint and executed an API call and I could not see `request` key in `kwargs`. So I think we need to go with `connexion.request.headers` .

![image](https://user-images.githubusercontent.com/106679486/184129504-94b9833e-fc05-4899-8d99-479416e707a8.png)